### PR TITLE
[nightwatch] fix screenshot command type

### PR DESCRIPTION
**Pull request by tim-yao**
Fix types based on this discussion https://discord.com/channels/618399631038218240/961886356693458964/1074663986886365186

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<[url here](https://discord.com/channels/618399631038218240/961886356693458964/1074663986886365186)>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## nightwatch ([<kbd>unpkg</kbd>](https://unpkg.com/browse/nightwatch@latest/))
was missing the following properties:
1. createClient
1. client
1. cli
1. runTests
1. CliRunner

<details>
<summary>as well as these 2 other properties...</summary>
<p>initClient, runner</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ddc28e4c090ac7ce3dbec223d72dc1c9208e9177
</p>

File: types/nightwatch/index.d.ts [hunk 1]
```diff
@@ -6683,7 +6683,10 @@ export interface WebDriverProtocolScreenCapture {
      * @example
      * browser.screenshot(true);
      */
-    screenshot(log_screenshot_data: boolean, callback?: (screenshotEncoded: string) => void): this;
+    screenshot(
+        log_screenshot_data: boolean,
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void,
+    ): Awaitable<this, string>;
```
Comment (garg3133) [inline types/nightwatch/index.d.ts:6689]: Actually, I looked at the code and found that `.screenshot` can be used in two ways, with and without `log_screenshot_data` parameter.

So, we can use the function overloading here to specify both ways:

```suggestion
    screenshot(
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void,
    ): Awaitable<this, string>;
    screenshot(
        log_screenshot_data: boolean,
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void,
    ): Awaitable<this, string>;
```

File: types/nightwatch/test/testClientCommands.ts [hunk 2]
```diff
@@ -297,6 +297,16 @@ describe('resizeWindow command demo', function() {
...
 
+//
+// .Screenshot
+//
+describe('screenshot command demo', function() {
+    test('async demo test', async function(browser) {
+        const result = await browser.screenshot(true);
+        isType<string>(result);
+    });
+});
```
Comment (garg3133) [inline types/nightwatch/test/testClientCommands.ts:308]: Can you please add a test for the sync way of using `.screenshot()` as well, with a callback?

File: types/nightwatch/index.d.ts (hunk 1 above)
Comment (garg3133) [reply types/nightwatch/index.d.ts:6689]: Please update the tests as well after making this change to include both the function overloads.

**Review by garg3133 — COMMENTED** [review]
Thanks for working on this PR! Below are a few thoughts:

File: types/nightwatch/index.d.ts (hunk 1 above)
Comment (tim-yao) [reply types/nightwatch/index.d.ts:6689]: Thanks @garg3133 . If `log_screenshot_data ` is optional, what about 
```suggestion
    screenshot(
        log_screenshot_data?: boolean,
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void,
    ): Awaitable<this, string>;
```
Is that providing the same as the function overloading but more simple?

Comment (tim-yao) [reply types/nightwatch/index.d.ts]: @garg3133 Thanks for the explanation. That makes sense to me.

**Review by garg3133 — APPROVED** [review]

**Comment by garg3133** [comment]
Ready to merge

**Pull request merged**